### PR TITLE
Implement basic route-id data type & use remix config.appDirectory.

### PR DESCRIPTION
**Pull request by dawnmist**
- Addresses [Suggestion]: Typechecking route id's #39
- Respects the remix config `appDirectory` setting
- Calculates correct relative path to where `config.appDirectory` resides based on `-o` output directory.
- Adds test with custom `config.appDirectory` path and multiple depth `-o` path.

### CODE REVIEW

**Comment by yesmeck** [comment]
@dawnmist thank you! Could you please fix the CI error?

**Comment by dawnmist** [comment]
It should be fixed now. :)

**Comment by dawnmist** [comment]
Is there a way to see what the actual Windows failures are?

File: packages/remix-routes/src/__tests__/cli.test.ts [hunk 1]
```diff
@@ -16,3 +16,10 @@ test('build v2 routes', async () => {
     fs.readFileSync(path.resolve(__dirname, '../../fixture/v2/node_modules/remix-routes.d.ts'), 'utf8'),
   ).toMatchSnapshot();
 });
+
+test("build customPaths routes", async () => {
+  await build(path.resolve(__dirname, "../../fixture/customPaths"), { outputDirPath: "./types/remix-routes", watch: false, strict: false });
+  expect(
+    fs.readFileSync(path.resolve( __dirname, "../../fixture/customPaths/types/remix-routes/remix-routes.d.ts"), "utf8"),
```
Comment (yesmeck) [inline packages/remix-routes/src/__tests__/cli.test.ts:23]: You can try printing out the content of remix-routes.d.ts to see what's generated on windows.

Comment (dawnmist) [reply packages/remix-routes/src/__tests__/cli.test.ts:32]: Next run should show that now.

I'm hoping it's not something like line-feed differences between linux & Windows.

File: packages/remix-routes/src/__tests__/cli.test.ts [hunk 2]
```diff
@@ -5,14 +5,30 @@ import { build } from '../cli';
 
 test('build v1 routes', async () => {
   await build(path.resolve(__dirname, '../../fixture/v1'), { outputDirPath: './node_modules', watch: false, strict: false });
+  console.log(fs.readFileSync(
+    path.resolve(__dirname, "../../fixture/v2/node_modules/remix-routes.d.ts"), "utf8"),
```
Comment (yesmeck) [inline packages/remix-routes/src/__tests__/cli.test.ts:9]:
```suggestion
    path.resolve(__dirname, "../../fixture/v1/node_modules/remix-routes.d.ts"), "utf8"),
```

File: packages/remix-routes/src/__tests__/cli.test.ts (hunk 1 above)
Comment (dawnmist) [reply packages/remix-routes/src/__tests__/cli.test.ts:32]: Argh. Sorry for the copy-paste error on the v1 route console log. Pushed the correct file path to log.

**Comment by yesmeck** [comment]
Great! CI is passing now. Could you please remove those temporary console.log statements, once that's done, we'll be ready to merge.

**Comment by dawnmist** [comment]
> Great! CI is passing now. Could you please remove those temporary console.log statements, once that's done, we'll be ready to merge.

I've done a rebase and removed the commit with the console logs. It should 🤞🏻 be good to go now.

**Comment by yesmeck** [comment]
Published as remix-routes@1.6.0. Cheers!

**Pull request merged**